src/main/java/com/cegeka/academy/service/mapper/UserChallengeMapper.java

```diff
@@ -0,0 +1,101 @@
+package com.cegeka.academy.service.mapper;
+
+import com.cegeka.academy.domain.Challenge;
+import com.cegeka.academy.domain.Invitation;
+import com.cegeka.academy.domain.UserChallenge;
+import com.cegeka.academy.service.dto.*;
+
+public class UserChallengeMapper {
+
+
+    public static UserChallenge convertUserChallengeDTOToUserChallenge(UserChallengeDTO userChallengeDTO){
+
+        if(userChallengeDTO != null){
+
+            UserChallenge userChallenge = new UserChallenge();
+            userChallenge.setId(userChallengeDTO.getId());
+            userChallenge.setUser(new UserMapper().userDTOToUser(userChallengeDTO.getUser()));
+            userChallenge.setInvitation(convertInvitationChallengeDTOTOInvitation(userChallengeDTO.getInvitation()));
+            userChallenge.setChallenge(convertChallengeDTOToChallenge(userChallengeDTO.getChallenge()));
```

Comment:
Aici daca `userChallengeDTO.getChallenge()` returneaza null, metoda `convertChallengeDTOToChallenge(...) `cred ca va crapa.  Also, daca am returna "null' cand ChallengeDTO este null, am avea un userChallenge fara Challenge. Daca vom incerca sa il introducem in baza de date va crapa pentru ca in dreptul campului Challenge exista @NotNull.